### PR TITLE
Update diagram in Developer Guide to separate parser

### PR DESCRIPTION
**Pull request by GilbertTan19**

### CODE REVIEW

File: docs/diagrams/ClearStatsParserSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+activate QuickCacheParser
+
+create ClearStatsCommandParser
+QuickCacheParser -> ClearStatsCommandParser
+activate ClearStatsCommandParser
+
+ClearStatsCommandParser -> QuickCacheParser
+deactivate ClearStatsCommandParser
+
+QuickCacheParser -> ClearStatsCommandParser : parse(" 1")
```
Comment (joshtyf) [inline docs/diagrams/ClearStatsParserSequenceDiagram.puml]: Is there supposed to be a whitespace for `" 1"`?

File: docs/diagrams/OpenParserSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+activate QuickCacheParser
+
+create OpenCommandParser
+QuickCacheParser -> OpenCommandParser
+activate OpenCommandParser
+
+OpenCommandParser -> QuickCacheParser
+deactivate OpenCommandParser
+
+QuickCacheParser -> OpenCommandParser : parse(" 1")
```
Comment (joshtyf) [inline docs/diagrams/OpenParserSequenceDiagram.puml]: is there supposed to be a whitespace at `" 1"`?

File: docs/diagrams/StatsParserSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+activate QuickCacheParser
+
+create StatsCommandParser
+QuickCacheParser -> StatsCommandParser
+activate StatsCommandParser
+
+StatsCommandParser -> QuickCacheParser
+deactivate StatsCommandParser
+
+QuickCacheParser -> StatsCommandParser : parse(" 1")
```
Comment (joshtyf) [inline docs/diagrams/StatsParserSequenceDiagram.puml:24]: is there supposed to be a whitespace at " 1"?

**Review by joshtyf — APPROVED** [review]
Looks good to me. I just have a few questions regarding whether or not there should be a whitespace in front of the indices. If there isn't supposed to be, please make the necessary changes. Else, can just merge

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -353,6 +361,9 @@ Step 2. The user executes `add q/ question... t/tag` command to add a flashcard
 <div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not be saved in the QuickCache, so the flashcard inside the QuickCache will not be updated.
 </div>
 
+The following sequence diagram shows how the Parser operation works:
```
Comment (FH-30) [inline docs/DeveloperGuide.md]: Parser here is inconsistent as it it capitalized

**Review by joshtyf — APPROVED** [review]
Alright great thanks!

**Pull request merged**